src/Stateless/Graph/StateGraph.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Stateless.Reflection;
+
+namespace Stateless.Graph
+{
+    /// <summary>
+    /// This class is used to generate a symbolic representation of the
+    /// graph structure, in preparation for feeding it to a diagram
+    /// generator 
+    /// </summary>
+    class StateGraph
+    {
+        public Dictionary<string, State> States { get; private set; } = new Dictionary<string, State>();
+        public List<Transition> Transitions { get; private set; } = new List<Transition>();
+        public List<Decision> Decisions { get; private set; } = new List<Decision>();
+
+        public StateGraph(StateMachineInfo machineInfo)
```

Comment:
This is a very long method, maybe it should be split into several smaller methods?